Sources/GRPCHTTP2Core/GRPCStreamStateMachine.swift

```diff
@@ -192,6 +192,23 @@ private enum GRPCStreamStateMachineState {
 
     var inboundMessageBuffer: OneOrManyQueue<[UInt8]>
 
+    /// This transition should only happen on the client-side.
+    /// It can happen if the request times out before the client outbound can be opened.
+    init(previousState: ClientIdleServerIdleState) {
+      self.maximumPayloadSize = previousState.maximumPayloadSize
+      // We don't need a framer or compressor since we won't be sending any messages.
+      self.framer = nil
```

Comment:
It's fine to just create one and not use it. In fact I'd rather we did that to avoid all the `!`s.